inverted_index.py

```diff
@@ -218,28 +222,31 @@ def MergeBlocks(self, total_num_blocks):
                 doc_ids_to_write = []
 
             if curr_lines_in_mem == self.MAX_LINES_TO_HOLD_IN_MEM:
-                for i in range(0, len(result_doc_ids_to_write)):
-                    self.WriteToFile(self.out_postings, result_doc_ids_to_write[i], True, write_posting_file_pointer)
-                    self.WriteToFile(self.out_dict, result_term_to_write[i], True, write_dict_file_pointer)
+                self.WriteToFile(self.out_postings, "".join(result_doc_ids_to_write), True, write_posting_file_pointer)
+                self.WriteToFile(self.out_dict, "".join(result_term_to_write), True, write_dict_file_pointer)
 
                 curr_lines_in_mem = 0
                 result_doc_ids_to_write = []
                 result_term_to_write = []
 
-
             term_to_write = curr_term
 
             for doc_id in curr_posting_list:
                 if len(doc_ids_to_write) == 0 or int(doc_id) != int(doc_ids_to_write[-1]):
                     doc_ids_to_write.append(doc_id)
 
-            # Check if can add next line in the block to queue
-            line = self.ReadFromFile("blocks/" + str(curr_block), blocks_offset[curr_block], read_file_pointers[curr_block])
-            # End of File
-            if line == '':
-                continue
-            blocks_offset[curr_block] = blocks_offset[curr_block] + len(line) + 1  # 1 for \n
-            q.put(QueueItem(line, curr_block))
+            # When all the lines from curr_block has been processed, we add in the next batch of
+            # lines from the block
+            if lines_per_block_mem[curr_block] == 0:
+                for li in range(0, lines_to_read_per_block):
+                    line = self.ReadFromFile("blocks/" + str(curr_block), blocks_offset[curr_block], read_file_pointers[curr_block])
```

Comment:
can extend ReadFromFile to also have a lines parameter, so that you don't have to loop it